demux/basemask.py

```diff
@@ -1,62 +1,94 @@
 # -*- coding: utf-8 -*-
-
-import sys
 import click
 import logging
 
-from pathlib import Path
+from path import Path
 import xml.etree.cElementTree as et
 
-from .utils import Samplesheet, HiSeqXSamplesheet, NIPTSamplesheet, HiSeq2500Samplesheet, MiseqSamplesheet
+from .utils import (
+    Samplesheet,
+    HiSeqXSamplesheet,
+    NIPTSamplesheet,
+    HiSeq2500Samplesheet,
+    MiseqSamplesheet,
+)
 
 log = logging.getLogger(__name__)
 
+
 @click.group()
 def basemask():
     """Samplesheet commands"""
     pass
 
+
 @basemask.command()
-@click.argument('rundir')
-@click.option('-l', '--lane', help='lane number')
-@click.option('-a', '--application', type=click.Choice(['wgs', 'wes', 'nipt', 'miseq']), help='sequencing type')
+@click.argument("rundir")
+@click.option("-l", "--lane", help="lane number")
+@click.option(
+    "-a",
+    "--application",
+    type=click.Choice(["wgs", "wes", "nipt", "miseq", "nova"]),
+    help="sequencing type",
+)
 def create(rundir, lane, application):
     """Create a basemask based on SampleSheet.csv and runParameters"""
 
+    # runParameters.xml
+    def parse_run_parameters(run_parameters_file):
+        return et.parse(Path(rundir).joinpath(run_parameters_file))
+
+    def create_basemask(run_params_tree, sheet):
+        read1_len = int(run_params_tree.findtext("Setup/IndexRead1"))
+        read2_len = int(run_params_tree.findtext("Setup/IndexRead2"))
+
+        lines = [line for line in sheet.lines_per_column("lane", lane)]
+
+        # get the index lengths
+        index1 = lines[0]["index"]
+        index2 = lines[0]["index2"] if "index2" in lines[0] else ""
```

Comment:
If demultiplexing has a constant file. 
EMPTY_STR = ""
I find this more readable, but is is more a question of style
```suggestion
        index2 = lines[0]["index2"] if "index2" in lines[0] else EMPTY_STR
```